asciidoc-confluence-publisher-maven-plugin/src/main/java/org/sahli/asciidoc/confluence/publisher/maven/plugin/AsciidocConfluencePublisherMojo.java

```diff
@@ -125,6 +140,10 @@ public void execute() throws MojoExecutionException {
             Map<String, Object> attributes = this.attributes != null ? this.attributes : emptyMap();
             ConfluencePublisherMetadata confluencePublisherMetadata = asciidocConfluenceConverter.convert(asciidocPagesStructureProvider, pageTitlePostProcessor, this.confluencePublisherBuildFolder.toPath(), attributes);
 
+            if (((this.username == null) || (this.password == null)) && this.mavenSettings != null) {
```

Comment:
In which case `mavenSettings` could actually be null?

Reply:
I guess you are right, this cannot happen. I can remove the check but on the other hand It does not hurt and is fail save.